### DISCUSSION TITLE
Porduktyzacja - wersjonowanie produktów

### DISCUSSION THREAD

**Discussion by kfijalko** [allegro/allegro-api#4489] [Suggestions - rejected]
:bulb: **Sugestia**

Obecnie tworzę produkty. Zagadnienie, które mnie nieco frapuje to "utrata kontroli nad danymi produktu".
Po wysłaniu produktu do bazy allegro, nie mam wpływu na jego dalsze losy.
Zgłoszenie uwagi do produktu przez formularz nie jest rozwiązaniem, ponieważ muszę wiedzieć czy coś zmieniliście w produkcie.
Przy założeniu, że mam kilkadziesiąt tysięcy produktów (części motoryzacyjne) odpytywanie o szczegóły produktu i porównywanie wszystkich danych jest nieefektywne. Również limity i blokady na metodach API nie pomagają.
No i teraz sygestia:
1) Wprowadzenie wersjonowania produktów.
Dla części tecdoc zwracacie "Wersja TecDoc  2106".
Czy moglibyście wprowadzić pole "wersja_danych" lub "data_aktalizacji_danych" dla produktów, które tworzą sprzedawcy.
Zmienialibyście to pole przy każdej edycji danych przez waszych pracowników.
2) Wprowadzenie dodatkowej metody, która na wejściu przyjmuje tablicę id produktów i zwraca pary product_id + wersja_danych

**Comment by hahyes** [comment]
Jest jeszcze "lepiej" ponieważ produkt może zostać bez ostrzeżenia usunięty i o powód usunięcia się nie doprosisz.
Dlatego dodałbym trzeci punkt: jeżeli usunięto, niech zwracany jest powód usunięcia produktu z katalogu.

**Comment by MartaNowaczyk** [comment]
Dziękuję za sugestię. Zgodnie z [naszym newsem](https://github.com/allegro/allegro-api/issues/4337) przenoszę wątek do zakładki [Discussions](https://github.com/allegro/allegro-api/discussions).

**Reply by hahyes** [reply]
Mam jeszcze teoretyczne pytanie nt. tworzenia produktów, jak już jest mowa o produktyzacji.
Załóżmy, że utworzyłem produkt, który został z jakiegoś powodu usunięty. Czy API wyrzuci mi błąd podczas dodawania tego samego produktu o duplikacie w katalogu produktów, czy też po usunięciu dane te są zwalniane i możliwe do ponownego dodania?

**Comment by MartaNowaczyk** [comment]
Po usunięciu jest możliwość ponownego dodania tego produktu, jeśli nie znajdziemy duplikatu.

**Comment by PrzemyslawLukanowski** [comment]
Dziękujemy za sugestię i opisanie potrzeby wersjonowania danych produktów. Zaktualizowaliśmy jej status na „odrzucona”. Na ten moment nie planujemy udostępniania w danych produktu informacji o wersji lub dacie aktualizacji danych.
